core/market/resolver/src/resolver/prop_parser.rs

```diff
@@ -209,14 +213,39 @@ named!(
 );
 
 named!(
-    string_literal<Literal>,
+    str_literal<Literal>,
     ws!(delimited!(
         char!('"'),
         do_parse!(val: take_until!("\"") >> (Literal::Str(str::from_utf8(val).unwrap()))),
         char!('"')
     ))
 );
 
+named!(
+    string_literal<Literal>,
+    ws!(delimited!(
+        char!('"'),
+        map!(
+            fold_many1!(
+                alt!(
+                    verify!(is_not!("\"\\"), |s: &[u8]| !s.is_empty())
+                        | preceded!(
+                            char!('\\'),
+                            alt!(
+                                value!(b"\\".as_bytes(), char!('\\'))
+                                    | value!(b"\"".as_bytes(), char!('"'))
+                            )
+                        )
+                ),
+                String::new(),
+                |acc, fragment| acc + str::from_utf8(fragment).unwrap()
+            ),
+            |str| Literal::String(str)
+        ),
+        char!('"')
+    ))
+);
+
```

Comment:
Maybe I'm missing something, but I don't see how `Literal::String` would be chosen over `Literal::Str` while parsing.

If we skip the unescaping part, the `Literal::String` type wouldn't be needed and parsing could be done via e.g.:

```rust
named!(
    string_literal<Literal>,
    ws!(do_parse!(
        val: delimited!(
            tag!("\""),
            escaped!(none_of!("\\\""), '\\', one_of!("\\\"")),
            tag!("\"")
        ) >> (Literal::Str(str::from_utf8(val).unwrap()))
    ))
);
```

Reply:
string_literal used here:
https://github.com/golemfactory/yagna/pull/1758/files#diff-b14d1e21873636609b0a7924e3830facdd507a38ad770f5f318c053774ca74acR446

I didn't find the way how I can parse string including escaped chars with nom crate version 2.*
I think it's not the best option, but acceptable workaround.

Another solution is upgrade nom and use something like this: https://docs.rs/nom/5.1.2/nom/combinator/fn.all_consuming.html in str_literal func

Reply:
I must admit I'd prefer going in the direction pointed by @mfranciszkiewicz above. 
I get the need of introducing `Literal::String()` but when it comes to parsing escaped quotes, use the simpler macro written by Marek, just "unescape" the result (by replacing \" with ").

Reply:
I have made the parser simpler. Workaround is removed. Unit test is added